app/serializers/search/postgres_comment_serializer.rb

```diff
@@ -0,0 +1,41 @@
+module Search
+  # TODO[@atsmith813]: Rename this to CommentSerializer once Elasticsearch is removed
+  class PostgresCommentSerializer < ApplicationSerializer
+    attribute :id, &:search_id
+
+    attributes :path, :public_reactions_count
+
+    attribute :body_text, &:body_markdown
+
+    attribute :class_name do |_comment, _params|
+      "Comment"
+    end
+
+    attribute :highlight do |comment, _params|
+      {
+        body_text: [comment.pg_search_highlight]
+      }
+    rescue PgSearch::PgSearchHighlightNotSelected
+      # This is needed because in Search::Postgres::Comment we only call the
+      # search if a term is provided. This means if a user searches with a
+      # blank term, we skip the line that executes .with_pg_search_highlight.
+      # Skipping this AND trying to call .pg_search_highlight raises an error
+      # which we ignore here - basically filling in highlights if they're
+      # there.
+    end
```

Comment:
A possible variation:

```suggestion
    attribute :highlight do |comment, params|
      if params[:search]
        {
          body_text: [comment.pg_search_highlight]
        }
      else
        {}
      end
    end
```

and then in `Search::Postgres::Comment`'s serializer you can pass a new param:

```ruby
.new(results, params: { users: users, search: term.present? }, is_collection: true)
```

This would cirumvent the need to catch an exception.

Reply:
I actually like that we explicitly catch the exception and document why. It's primarily because 1) that's how the gem works and 2) we're using a serializer.

The change you proposed feels more complicated. I could see us coming back in the future and asking ourselves, _"Why are we passing in `search: boolean`? Why is this set up this way?"_. And the implementation you've suggested hides all of that context.

While catching exceptions isn't necessarily great, I feel like it's much more clear as to _what_ is going on and _why_.

I can also see a future where if we do this practice now, we'll get into a habit of passing in all sorts of custom attributes so the serializer can make more "decisions". I was hoping to keep the serializers as simple as possible. I could see our future selves getting a little lost with the logic spread between the serializer and the service and the controller action calling the service.

Can we leave it as is?

Reply:
Sure!